### PR TITLE
Clarify active runtime ID usage in the executor

### PR DESCRIPTION
**Pull request by stnguyen90**
The runtimeId will be the ID passed into the executor. From there, the runtimeName will be the hostname + runtimeId. We want to prefix with hostname to support multiple executors on the same machine.

The runtimeName will be used for the container name and the active runtimes table.

Confirmed building works:

image

Confirmed executing works:

image

### CODE REVIEW

File: tests/ExecutorTest.php [hunk 1]
```diff
@@ -152,12 +152,12 @@ public function testBuild(): array
...
         $this->assertEquals(200, $response['headers']['status-code']);
         $this->assertEquals(1, count($response['body']));
-        $this->assertEquals('test-build', $response['body'][0]['name']);
+        $this->assertStringContainsString('test-build', $response['body'][0]['name']);
 
         /** Get runtime */
         $response = $this->client->call(Client::METHOD_GET, '/runtimes/test-build', [], []);
         $this->assertEquals(200, $response['headers']['status-code']);
-        $this->assertEquals('test-build', $response['body']['name']);
+        $this->assertStringContainsString('test-build', $response['body']['name']);
```
Comment (Meldiron) [inline tests/ExecutorTest.php:160]: If there is stringEndsWith, lets use that. Will make it clear there is prefix we ignore. If not, all good.

File: app/http.php [hunk 2]
```diff
@@ -718,10 +716,9 @@ function (string $runtimeId, ?string $payload, string $path, string $method, arr
                 $payload = '';
             }
 
-            $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
-            $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
+            $runtimeName = System::getHostname() . '-' . $runtimeId;
 
-            $log->addTag('runtimeId', $activeRuntimeId);
+            $log->addTag('runtimeId', $runtimeId);
```
Comment (Meldiron) [inline app/http.php:721]: Lets add log for runtimeName as well

File: app/http.php [hunk 3]
```diff
@@ -674,17 +673,16 @@ function removeAllRuntimes(Table $activeRuntimes, Orchestration $orchestration):
     ->inject('response')
     ->inject('log')
     ->action(function (string $runtimeId, Orchestration $orchestration, Table $activeRuntimes, Response $response, Log $log) {
-        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
-        $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
+        $runtimeName = System::getHostname() . '-' . $runtimeId;
 
-        $log->addTag('runtimeId', $activeRuntimeId);
+        $log->addTag('runtimeId', $runtimeId);
```
Comment (Meldiron) [inline app/http.php:678]: Lets log runtimeName instead

File: app/http.php [hunk 4]
```diff
@@ -651,15 +650,15 @@ function removeAllRuntimes(Table $activeRuntimes, Orchestration $orchestration):
     ->inject('response')
     ->inject('log')
     ->action(function (string $runtimeId, Table $activeRuntimes, Response $response, Log $log) {
-        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
+        $runtimeName = System::getHostname() . '-' . $runtimeId;
 
-        $log->addTag('runtimeId', $activeRuntimeId);
+        $log->addTag('runtimeId', $runtimeId);
```
Comment (Meldiron) [inline app/http.php:655]: Lets log runtimeName instead

File: app/http.php [hunk 5]
```diff
@@ -390,15 +390,14 @@ function removeAllRuntimes(Table $activeRuntimes, Orchestration $orchestration):
...
     ->inject('log')
     ->action(function (string $runtimeId, string $image, string $entrypoint, string $source, string $destination, array $variables, string $runtimeEntrypoint, string $command, int $timeout, bool $remove, int $cpus, int $memory, string $version, Orchestration $orchestration, Table $activeRuntimes, Response $response, Log $log) {
-        $activeRuntimeId = $runtimeId; // Used with Swoole table (key)
-        $runtimeId = System::getHostname() . '-' . $runtimeId; // Used in Docker (name)
+        $runtimeName = System::getHostname() . '-' . $runtimeId;
 
         $runtimeHostname = \uniqid();
 
-        $log->addTag('runtimeId', $activeRuntimeId);
+        $log->addTag('runtimeId', $runtimeId);
```
Comment (Meldiron) [inline app/http.php:397]: Lets log runtimeName instead

File: app/http.php [hunk 6]
```diff
@@ -486,13 +485,13 @@ function removeAllRuntimes(Table $activeRuntimes, Orchestration $orchestration):
...
                 image: $image,
-                name: $runtimeId,
+                name: $runtimeName,
                 hostname: $runtimeHostname,
                 vars: $variables,
                 command: $runtimeEntrypointCommands,
                 labels: [
                     'openruntimes-executor' => System::getHostname(),
-                    'openruntimes-runtime-id' => $activeRuntimeId
+                    'openruntimes-runtime-id' => $runtimeId
```
[the remark this answers is not in the document]
Comment (stnguyen90) [reply app/http.php:494]: I have confirmed:

* [x] the maintenance task will remove inactive containers 
   image
* [x] the executor shuts down containers when it shuts down
  image

File: tests/ExecutorTest.php [hunk 7]
```diff
@@ -152,12 +152,12 @@ public function testBuild(): array
         $response = $this->client->call(Client::METHOD_GET, '/runtimes', [], []);
         $this->assertEquals(200, $response['headers']['status-code']);
         $this->assertEquals(1, count($response['body']));
-        $this->assertEquals('test-build', $response['body'][0]['name']);
+        $this->assertStringContainsString('test-build', $response['body'][0]['name']);
```
Comment (Meldiron) [inline tests/ExecutorTest.php:155]:
```suggestion
        $this->assertStringEndsWith('test-build', $response['body'][0]['name']);
```

**Review by Meldiron — APPROVED** [review]

**Pull request merged**